python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -107,18 +108,23 @@ class DiffMatchedTractCatalogConnections(
         dimensions=("tract", "skymap"),
     )
     cat_matched = cT.Output(
-        doc="Catalog with reference and target columns for matched sources only",
+        doc="Catalog with reference and target columns for joined sources",
         name="matched_{name_input_cat_ref}_{name_input_cat_target}",
-        storageClass="DataFrame",
+        storageClass="ArrowAstropy",
         dimensions=("tract", "skymap"),
     )
     diff_matched = cT.Output(
         doc="Table with aggregated counts, difference and chi statistics",
         name="diff_matched_{name_input_cat_ref}_{name_input_cat_target}",
-        storageClass="DataFrame",
+        storageClass="ArrowAstropy",
```

Comment:
After having spoken with @taranu, it seems like these should be rolled back to `DataFrame` storage classes for the time being.

Reply:
PS - please do run some test data through post-change to make sure that everything still works as expected, e.g., no missing `.iloc`s anywhere!